src/04-select.md

```diff
@@ -1,4 +1,92 @@
+- `sendt`: This is a list of cosock sockets that are waiting to be ready to `send`
+- `timeout`: This is the maximum amount of seconds to wait for one or more entries in `recvt` or `sendt` to be ready
+  - If this value is `nil` or negative it will treat the timeout as infinity
+
+> Note: The list entries for `sendt` and `recvt` can be other "cosock aware" tables like the
+> [lustre WebSocket](https://github.com/cosock/lustre), for specifics on how to make a table "cosock aware" 
+> [see the chapter on it](07-setwaker.md)
+
+Its return values are
+
+- `recvr`: A list of ready receivers, any entry here should be free to call `receive` and
+           immediately be ready
+- `sendr`: A list of ready senders, any entry here should be free to call `send` and immediately be
+           ready
+- `err`: If this value is not `nil` it represents an error message
+  - The most common error message here would be `"timeout"` if the `timeout` argument provided
+    is not `nil` and positive
+
+So, how would we use something like this? Let's consider our `clients_server.lua` example
+from the [spawn chapter](./02-spawn.md), where we called `cosock.spawn` every time a new
+client was `accept`ed, this works but we don't have much control over how many tasks we end
+up spawning. In large part, this is because we don't know how long each task will run. Ideally,
+we would be able to handle all of the client connections on the same task as the server and to
+do that, we can use `select`.
+
+```lua
+{{ #include ../examples/clients_server_select.lua }}
+```
+
+The above is an updated version of our
+[clients/server example](./02-spawn.md#clientsserver-example) with some updates to limit
+the total number of connections to 5, let's go over the changes.
+
+First, we've added a few helper functions to handle the different events in our system,
+the first is for when a client connection is ready to receive `handle_recv` takes 2 arguments,
+`client` which is a `cosock.socket.tcp` that was returned from a call
+to `accept` and `clients` which is a table where the keys are `cosock.socket.tcp` clients
+and the values are `true`. We first call `client:recieve` to get the bytes from
+the client and if that returns a string that contains `"ping"` then we send our
+`"pong"` message. There are few places where this can go wrong, the call to `receive`
+could return `nil` and an error message or not `"ping"` or the call to `send` could
+return `nil` and an error message; if the error message is `"closed"` or the request
+didn't contain `"ping"` then we want to remove `client` from `clients` and if it was
+the latter then we want to call `client:close`.
+
+Next up we have `handle_accept` this also takes 2 arguments `server` which is a 
+`cosock.socket.tcp` socket that is listening and the same map of `clients`. If a
+call to `accept` returns a `client` then we add that `client` into our `clients` map.
+If `accept` returns `nil` and `err` isn't `"timeout"` then we raise an error.
```

Comment:
This description doesn't exactly match the code.  `handle_accept` doesn't return and handles adding to the client list itself.